onnxscript/converter.py

```diff
@@ -759,7 +761,32 @@ def _cast_like_binary_expression(self, op, left, right):
         schema = op.get_schema()
         return autocast.static_cast_inputs(self, schema, left, right)
 
-    def translate_bin_op_expr(self, node):
+    def translate_bool_op_expr(self, node: ast.BoolOp) -> ConverterExpression:
+        if isinstance(node.op, ast.And):
+            op = values.Op(self.default_opset, "And")
+        elif isinstance(node.op, ast.Or):
+            op = values.Op(self.default_opset, "Or")
+        else:
+            raise ValueError(self.message(node, f"Unsupported operator {node.op!r}."))
+
+        expr: Optional[ConverterExpression] = None
+        operands = iter(node.values)
+        try:
+            while True:
+                left, right = self._cast_like_binary_expression(
+                    op,
+                    expr if expr else self.translate_expr(next(operands)),
+                    self.translate_expr(next(operands)),
+                )
+                ovar = self.generate_unique_name()
+                self.emit([ovar], op, [left, right], [])
+                expr = ConverterExpression(ovar, ConverterExpressionKind.ANY)
+        except StopIteration:
```

Comment:
Not sure what the goal is. How about:
```py
   expr = node.values[0]
   for operand in node.values[1:]:
      ...
```
If the concern is about creating a copy of the list (I am not sure how the compilers implement this), it doesn't seem like it would be a significant perf concern.

Reply:
Just the `while True` loop making me a little nervous

Reply:
Yeah, that's just Python not being my first language shining through, so I hadn't thought of slicing. The point was that in the first pass, we need to handle the first two operands, hence calling `next` on the iterater twice. Updated.